pkg/discovery/monitoring/kubelet/kubelet_monitor.go

```diff
@@ -59,66 +59,56 @@ func (m *KubeletMonitor) GetMonitoringSource() types.MonitoringSource {
 
 func (m *KubeletMonitor) ReceiveTask(task *task.Task) {
 	m.reset()
-	m.nodeList = task.NodeList()
+	m.node = task.Node()
 }
 
-func (m *KubeletMonitor) Do(stopChan <-chan struct{}) *metrics.EntityMetricSink {
+func (m *KubeletMonitor) Do() (*metrics.EntityMetricSink, error) {
 	glog.V(4).Infof("%s has started task.", m.GetMonitoringSource())
-	err := m.RetrieveResourceStat(stopChan)
+	err := m.RetrieveResourceStat()
 	if err != nil {
 		glog.Errorf("Failed to execute task: %s", err)
+		return m.metricSink, err
 	}
 	glog.V(4).Infof("%s monitor has finished task.", m.GetMonitoringSource())
-	return m.metricSink
+	return m.metricSink, nil
 }
 
-// RetrieveResourceStat retrieves resource stats for the received list of nodes.
-func (m *KubeletMonitor) RetrieveResourceStat(stopChan <-chan struct{}) error {
-	if m.nodeList == nil || len(m.nodeList) == 0 {
-		return errors.New("empty node list")
+// RetrieveResourceStat retrieves resource stats for the received node.
+func (m *KubeletMonitor) RetrieveResourceStat() error {
+	if m.node == nil {
```

Comment:
nit: this check could be caught earlier in Do() or even before

Reply:
Updated, PTAL.